src/main/java/team05/integrated_feed_backend/module/post/repository/impl/PostRepositoryImpl.java

```diff
@@ -0,0 +1,28 @@
+package team05.integrated_feed_backend.module.post.repository.impl;
+
+import org.springframework.stereotype.Repository;
+
+import com.querydsl.jpa.impl.JPAQueryFactory;
+
+import lombok.RequiredArgsConstructor;
+import team05.integrated_feed_backend.module.post.entity.Post;
+import team05.integrated_feed_backend.module.post.entity.QPost;
+import team05.integrated_feed_backend.module.post.repository.PostRepositoryCustom;
+
+@Repository
+@RequiredArgsConstructor
+public class PostRepositoryImpl implements PostRepositoryCustom {
+
+	private final JPAQueryFactory jpaQueryFactory;
+
+	@Override
+	public Post findDetailPostById(Long id) {
+
+		QPost post = QPost.post;
+		return jpaQueryFactory
+			.selectFrom(post)
+			.where(post.postId.eq(id))
+			.fetchOne();
```

Comment:
❗️ 혹시 `fetchJoin()` 을 안하신 이유가 있으실까요? 현재는 Spring Data JPA의 `findById(Long id)` 와 차이점이 크게 없는 것 같아서요!

Reply:
```suggestion
	@Override
	public Optional<Post> findDetailPostById(Long id) {

		QPost post = QPost.post;
		QPostHashtag postHashtag = QPostHashtag.postHashtag;
		QHashtag hashtag = QHashtag.hashtag1;

		return Optional.ofNullable(
			jpaQueryFactory
				.selectFrom(post)
				.leftJoin(post.postHashtags, postHashtag).fetchJoin()
				.leftJoin(postHashtag.hashtag, hashtag).fetchJoin()
				.where(post.postId.eq(id))
				.fetchOne()
		);
		
	}
```

아! 이 부분 지적 감사합니다! 쿼리는 처음이라, 차이점이 뭔지 고민했는데, 배워갑니다 👍